src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsResource.cs

```diff
@@ -45,10 +45,43 @@ public class AzureEventHubsResource(string name, Action<AzureResourceInfrastruct
     /// <summary>
     /// Gets the connection string template for the manifest for the Azure Event Hubs endpoint.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression =>
-        IsEmulator
-        ? ReferenceExpression.Create($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;")
-        : ReferenceExpression.Create($"{EventHubsEndpoint}");
+    public ReferenceExpression ConnectionStringExpression => BuildConnectionString();
+
+    private ReferenceExpression BuildConnectionString()
+    {
+        var builder = new ReferenceExpressionBuilder();
+
+        if (IsEmulator)
+        {
+            // ConnectionString: Endpoint=...
+            builder.Append($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true");
+        }
+        else
+        {
+            // FQNS: Uri format, e.g. https://...
+            builder.Append($"{EventHubsEndpoint}");
+        }
+
+        if (!Hubs.Any(hub => hub.IsDefaultEntity))
+        {
+            // Of zero or more hubs, none are flagged as default
+            return builder.Build();
+        }
+
+        // Of one or more hubs, only one may be flagged as default
+        var defaultEntity = Hubs.Single(hub => hub.IsDefaultEntity);
```

Comment:
Can't we use `SingleOrDefault` and only do 1 check through the loop?

Also, we can do this before the above check, and only have a single `IsEmulator` check.

Reply:
SingleOrDefault will return null for two or more event hubs.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.singleordefault?view=net-9.0

SingleOrDefault throws InvalidOperationException when the input sequence contains more than one element.

Reply:
Ah, sorry -- I was misremembering the reasons why I didn't use this pattern. Yeah, I _expected_ it to return `default` if not single, but you're right -- it throws. You're right that it did need another pass to consolidate the logic. I just pushed another commit to address this.

Reply:
Done.